src/Internal/Payment/Router.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class Router
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WCPay\Core\Server\Request\Get_Payment_Process_Factors;
+use WCPay\Database_Cache;
+use WCPay\Exceptions\API_Exception;
+use WCPay\Internal\Payment\Factor;
+
+/**
+ * Until the new payment process is fully developed, and the legacy
+ * process is gone, the new process is managed as a feature behind a router.
+ *
+ * This class will be removed once the new process is the default
+ * option for all payments.
+ */
+class Router {
+	/**
+	 * Database cache.
+	 *
+	 * @var Database_Cache
+	 */
+	protected $database_cache;
+
+	/**
+	 * Class constructor, receiving dependencies.
+	 *
+	 * @param Database_Cache $database_cache Database cache.
+	 */
+	public function __construct( Database_Cache $database_cache ) {
+		$this->database_cache = $database_cache;
+	}
+
+	/**
+	 * Checks whether a given payment should use the new payment process.
+	 *
+	 * @param Factor[] $factors Factors, describing the type and conditions of the payment.
+	 * @return bool
+	 * @psalm-suppress MissingThrowsDocblock
+	 */
+	public function should_use_new_payment_process( array $factors ): bool {
+		$allowed_factors = $this->get_allowed_factors();
+
+		// This would make sure that the payment process is a factor as well.
+		$factors[] = Factor::NEW_PAYMENT_PROCESS();
```

Comment:
It’s not controlling whether the new process gets entered regardless of the other factors, rather when they are missing. The point is that if no other factor is present, the `foreach` loop would never be entered, and the method would immediately return `true` for the simplest payments.

Even if this method is coded differently, we still need the factor to be controlled by the server, and that’s actually going to be the first factor that we enable, rather than the last (global) one. 

I see how that could be misleading, I’ll update the comment.

Reply:
My thinking was that this factor must always be present, which is why I added it to this method, to have it centralised and impossible to miss. However, adding it to the gateway too will not hurt. I’ll do so on Monday.

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/7035/commits/b029106a7bf3160b08a9ccf27b93cd9653bfb2b7